Sources/Tracking/CustomerIOImplementation.swift

```diff
@@ -83,6 +88,8 @@ class CustomerIOImplementation: CustomerIOInstance {
         identifier: String,
         body: RequestBody
     ) {
+        CIODataPipeline.analytics.identify(userId: identifier, traits: jsonAdapter.toJson(body))
```

Comment:
Should this line occur farther down in this function body? 

For example, the next line of the function is `if identifier.isBlankOrEmpty() { return }`. I was expecting that we only forward the identify request to the CDP module after we run logic such as this?

Reply:
I second Levi. The following line of code (`if identifier.isBlankOrEmpty() ...`) seems to have no value if we identify the user as the first thing in the function. This could be confusing as the user could be identified anonymously and then see the error log in the console `profile cannot be identified: Identifier is empty. Please retry with a valid, non-empty identifier` in case identifier is blank or empty though a corner case but possible.

Either we re-think the implementation or we make sure if with segment do we need to check identifier for blank or empty ?

Reply:
Agreed, this line `if identifier.isBlankOrEmpty()` in question would ideally be removed with the removal of tracking code. So we might not even cater this case. I am not removing any code in these PRs to avoid confusion. 

In-case we decide to keep this check for some reason, because Segment doesn't validate for empty strings then we can put this line below.

Reply:
Sounds good. Thanks for the clarification, Shahroz! Would you want to add a `TODO` here as well to not miss out removing this statement in the future tickets ?

Reply:
I can understand that. 

I would be OK with returning early and ignoring the request as long as we also log a warning to help the customer understand they may have done something wrong.